models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+    # and find the item that matches the PFT
+    for (jd_name in names(envcanopy_jsondata)) {
+      # First make sure that we are looking at the pftN entries, not
+      # comments, or other non-PFT variables
+      if (grepl("pft", jd_name)) {
+        cur_pft_name <- envcanopy_jsondata[[jd_name]]$name
+
+        # Then pick out only the entry for the correct pft
+        if (identical(cur_pft_name, pftname)) {
+          print(cur_t_name)
+          print(tname)
+          if (tname == "extinction_coefficient_diffuse") {
+            envcanopy_jsondata[[jd_name]]$er = trait_df[[cur_t_name]]
+          } else if (tname == "SW_albedo") {
+            envcanopy_jsondata[[jd_name]]$albvisnir = trait_df[[cur_t_name]] 
+          } else if (tname == "cuticular_cond") {
+            envcanopy_jsondata[[jd_name]]$gl_c = trait_df[[cur_t_name]]
+          } else if (tname == "gcmax") {
+            envcanopy_jsondata[[jd_name]]$glmax = trait_df[[cur_t_name]]
+          } else {
+            # pass...variable not in this file or datastructure
+            PEcAn.logger::logger.info("pass...variable not in this file or datastructure")
+          }
+        }
+      }
+    } # end loop over json for env canopy
+  }
+
+  # Write it back out to disk (overwriting ok??)
+  dimveg_exportJson <- toJSON(dimveg_jsondata)
+  write(dimveg_exportJson, "/tmp/dimveg_newfile.json")
+
+  envcanopy_exportJson <- toJSON(envcanopy_jsondata)
+  write(envcanopy_exportJson, "/tmp/envcanopy_newfile.json")
+
+
+  # (3)
+  # Format a new dvmdostem parameter file using the new json file as a source.
+
+  if (dir.exists(file.path(rundir, "parameters/"))) {
+    # pass
+  } else {
+    print("No parameter/ directory in run directory! Need to create...")
+    dir.create(file.path(rundir,"parameters" ))
+  }
+  ref_file <- paste0(file.path(appbinary_path, "parameters/"), 'cmt_dimvegetation.txt')
+  new_param_file <- paste0(file.path(rundir, "parameters/"), "cmt_dimvegetation.txt")
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--fmt-block-from-json", "/tmp/dimveg_newfile.json", ref_file)),
+          stdout=new_param_file, wait=TRUE)
+
+  ref_file <- paste0(file.path(appbinary_path, "parameters/"), 'cmt_envcanopy.txt')
+  new_param_file <- paste0(file.path(rundir, "parameters/"), "cmt_envcanopy.txt")
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--fmt-block-from-json", "/tmp/envcanopy_newfile.json", ref_file)),
+          stdout=new_param_file, wait=TRUE)
+
+
+  # TODO:
+  #  [x] finish with parameter update process
+  #  [x] dynamically copy parameters to right place
+  #  - dynamically copy the output_spec from insts folder to the
+  #    right place (see variable above for getting stuff from inst)
+  #  - figure out how to handle the met.
+  #     -> step one is symlink from raw data locations (Model install folder)
+  #        into pecan run folder, maybe do this within job.sh?
```

Comment:
If you symlink to the met location, then yes you need to do that in job.sh since the model and met may be on a remote server, and thus you can't create that link right now. If you copy the met then you can do that now since for remote runs the whole rundir will be rsynced to the remote machine.